include/wx/validate.h

```diff
@@ -17,9 +17,44 @@
 
 #include "wx/event.h"
 
+class WXDLLIMPEXP_FWD_CORE wxValidator;
 class WXDLLIMPEXP_FWD_CORE wxWindow;
 class WXDLLIMPEXP_FWD_CORE wxWindowBase;
 
+// ----------------------------------------------------------------------------
+// wxValidationEvent
+// ----------------------------------------------------------------------------
+
+class WXDLLIMPEXP_CORE wxValidationEvent : public wxCommandEvent
+{
+public:
+    wxValidationEvent() {}
+    wxValidationEvent(wxValidator *validator, wxEventType type, wxWindow *win);
+
+    void SetErrorMessage(const wxString& errormsg) { SetString(errormsg); }
+    wxString GetErrorMessage() const { return GetString(); }
+
+    void SetErrorCode(long code) { SetExtraLong(code); }
```

Comment:
Do we really need to use `long` error codes? What are they going to be? I think we should either define some enum with common errors or maybe just remove this completely and rely on the user-readable error message because I don't really know what difference can it make to the program to know why exactly has the validation failed -- in any case, the user needs to correct the entry.

Reply:
Good idea. the user already knows what this entry is for. and it's enough for him to know the boolean state of the entry (i.e ok or invalid) to come with a suitable error message.